walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,82 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.LinkedList;
+import java.util.Comparator;
 
 public final class FindMeetingQuery {
+  List<TimeRange> busyTimes = new ArrayList<>();
+  List<TimeRange> freeTimes = new ArrayList<>();
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    busyTimes = getBusyTimes(events, request);
+    List<TimeRange> overlappingTimes = findOverlappingTimes();
+    return (getFreeTimes(overlappingTimes, request));
+  }
+
+  // Compare attendees of given events with attendees for requested meeting to determine conflicts
+  private List<TimeRange> getBusyTimes(Collection<Event> events, MeetingRequest request) {
+
+    Collection<String> attendees = request.getAttendees();
+    for (Event event : events) {
+      if(!(Collections.disjoint(event.getAttendees(), attendees))) {
+        busyTimes.add(event.getWhen());
+      }
+    }
+    Collections.sort(busyTimes, TimeRange.ORDER_BY_START);
+    return busyTimes;
+  }
+
+  // Returns list of consolidated busy time ranges accounting for overlapping events
+  private List<TimeRange> findOverlappingTimes() {
+    List<TimeRange> overlappingTimes = new ArrayList<>();
+    int tempStartTime = -1;
+    int tempEndTime = 0;
+    for (TimeRange timeRange : busyTimes) {
```

Comment:
Why are tempStart/EndTime needed? What are they intended to represent?

Reply:
They are used to keep track of information from time ranges that are not currently being looped through. For example, if there are two timeranges, 8-10 and 9-10, I would want to keep track of the start time of 8:00 so that I can return the time range 8-10. Let me know if that's unclear and I can provide more clarification!

Reply:
Thanks for the clarifying this. I suggest two things to make the code clearer: 1) variable names that convey more context 2) A comment either right above the variable definitions, or in the javadoc for this method.

Reply:
I changed the variable names and added more comments to help clarify a bit of the logic and naming. If you think there's a better name for the variables, please let me know!

Reply:
That works. Thank you.